tensorflow_probability/python/distributions/ordered_logistic_test.py

```diff
@@ -0,0 +1,156 @@
+
+    for test in ["cutpoints", "location", "both"]:
+
+      for batch_shape in ([], [1], [1, 2, 3]):
+
+        if test == "cutpoints":
+          cutpoints = tf.broadcast_to(
+              tf.constant([1., 2.], dtype=tf.float64), batch_shape + [2])
+          location = tf.constant(0., dtype=tf.float64)
+        elif test == "location":
+          cutpoints = tf.constant([1., 2.], dtype=tf.float64)
+          location = self._rng.randn(*batch_shape)
+        elif test == "both":
+          cutpoints = tf.broadcast_to(
+              tf.constant([1., 2.], dtype=tf.float64), batch_shape + [2])
+          location = self._rng.randn(*batch_shape)
+
+        dist = tfd.OrderedLogistic(cutpoints, location)
+
+        self.assertAllEqual(dist.batch_shape, batch_shape)
+        self.assertAllEqual(
+            self.evaluate(dist.batch_shape_tensor()), batch_shape)
+
+        self.assertAllEqual(dist.event_shape, [])
+        self.assertAllEqual(self.evaluate(dist.event_shape_tensor()), [])
+
+        log_probs_shape = tf.shape(dist.categorical_log_probs())
+        self.assertAllEqual(self.evaluate(log_probs_shape), batch_shape + [3])
+
+        sample_shape = tf.shape(dist.sample(seed=test_util.test_seed()))
+        self.assertAllEqual(self.evaluate(sample_shape), batch_shape)
+
+        sample_shape_n = tf.shape(
+            dist.sample([4, 5], seed=test_util.test_seed()))
+        self.assertAllEqual(self.evaluate(sample_shape_n), [4, 5] + batch_shape)
+
+  def testProbs(self):
+
+    # survival functions
+    # P(Y > 0) = sigmoid(1) = 0.7310586
+    # P(Y > 1) = sigmoid(0) = 0.5
+    # P(Y > 2) = sigmoid(-1) = 0.26894143
+
+    # probs
+    # P(Y = 0) = 1. - sigmoid(1) = 0.2689414
+    # P(Y = 1) = sigmoid(1) - sigmoid(0) = 0.2310586
+    # P(Y = 2) = sigmoid(0) - sigmoid(-1) = 0.23105857
+    # P(Y = 3) = sigmoid(-1) = 0.26894143
+    expected_probs = [0.2689414, 0.2310586, 0.23105857, 0.26894143]
+    dist = tfd.OrderedLogistic([-1., 0., 1.], 0.)
+
+    categorical_probs = self.evaluate(dist.categorical_probs())
+    self.assertAllClose(expected_probs, categorical_probs, atol=1e-6)
+
+    probs = np.flip(self.evaluate(dist.prob([3, 2, 1, 0])))
+    self.assertAllClose(expected_probs, probs, atol=1e-6)
+
+  def testMode(self):
+    # 2 cutpoints i.e. 3 possible outcomes. 3 "batched" distributions with the
+    # logistic distribution location well within the large cutpoint spacing so
+    # mode is obvious
+    dist = tfd.OrderedLogistic([-10., 10.], [-20., 0., 20.])
+    mode = self.evaluate(dist.mode())
+    self.assertAllEqual([0, 1, 2], mode)
+
+  def testSample(self):
+    # as per `testProbs`
+    dist = tfd.OrderedLogistic([-1., 0., 1.], 0.)
+    samples = self.evaluate(dist.sample(100000, seed=test_util.test_seed()))
+    expected_probs = [0.2689414, 0.2310586, 0.23105857, 0.26894143]
+    for k, p in enumerate(expected_probs):
+      self.assertAllClose(np.mean(samples == k), p, atol=0.01)
+
+  def testEntropy(self):
```

Comment:
Perhaps add a test that computes entropy via sampling and compares to the analytic entropy?

Same with KL divergence.

Reply:
have added a basic test for this in https://github.com/tensorflow/probability/pull/753/commits/95e03d8c11e9c643a47cef7e2e22f804d69b1568 went for 1e5 samples and a tolerance of 0.01 (a bit of a guess and seemed to pass first time) hope that is OK.